### PR TITLE
Async create

### PR DESCRIPTION
**Pull request by ProgrammerDan**
Covers /nlcg , /nl gui create , and newfriend join w/ cascading name index seeking.

tested, will go live on devoted shortly, I'll update this if any issues appear.

Note that everything that needs to be synchronous _is_ and everything that can be asynchronous _should be_.

### CODE REVIEW

File: namelayer-spigot/src/main/java/vg/civcraft/mc/namelayer/GroupManager.java [hunk 1]
```diff
@@ -49,6 +49,82 @@ public int createGroup(Group group){
...
+	}
+	
+	private void doCreateGroupAsync(final Group group, final RunnableOnGroup postCreate) {
+		GroupCreateEvent event = new GroupCreateEvent(
+				group.getName(), group.getOwner(),
+				group.getPassword());
+		Bukkit.getPluginManager().callEvent(event);
+		if (event.isCancelled()){
+			NameLayerPlugin.log(Level.INFO, "Group create was cancelled for group: " + group.getName());
+			postCreate.setGroup(new Group(group.getName(), group.getOwner(), true, group.getPassword(), -1));
```
Comment (rourke750) [inline namelayer-spigot/src/main/java/vg/civcraft/mc/namelayer/GroupManager.java]: Can you comment here why a new group object is created from the original.

Comment (ProgrammerDan) [inline namelayer-spigot/src/main/java/vg/civcraft/mc/namelayer/GroupManager.java]: Same process as before, no idea why it was done before just didn't change it.

**Comment by ProgrammerDan** [comment]
Perfect. Live testing on devoted in under an hour, I'll  update this PR if things come up.

**Comment by ProgrammerDan** [comment]
No issues on devoted. Merging and staging.

**Pull request merged**